src/core/net/ip6.cpp

```diff
@@ -512,7 +513,11 @@ void Ip6::HandleSendQueue(void)
     while ((message = mSendQueue.GetHead()) != nullptr)
     {
         mSendQueue.Dequeue(*message);
-        IgnoreError(HandleDatagram(*message, kFromHostAllowLoopBack));
+        if (message->GetOrigin() != Message::kOriginHostUntrusted)
+        {
+            message->SetOrigin(Message::kOriginHostAllowLoopBack);
+        }
```

Comment:
Do we want to have this check and call `SetOrigin()` call here?
I think we can expect that the `message` should already have its origin set properly by whatever module created/allocated it.

We use `kFromHostAllowLoopBack` before since `ip6.cpp` was the only module tracking/using origin, but now it is directly managed by the module allocating the `message`.

Reply:
I updated the code to set the origin everywhere before enqueuing the packet.